ndinterp/src/grid/cubic.rs

```diff
@@ -1,35 +1,29 @@
 //! Implements cubic interpolation algorithms
+//!
+//! These are the algorithms used by the LHAPDF library for `alpha_s` and pdf(x, q)
+//!
 
 use crate::grid::{Grid, GridSlice};
 use crate::interpolate::InterpolationError;
 pub use crate::interpolate::Interpolator;
-use ndarray::{Axis, Ix1, Ix2};
+use itertools::izip;
+use ndarray::{Axis, Dimension, Ix1, Ix2, Slice};
+
+/// Cubic interpolation
+#[derive(Debug)]
+pub struct Cubic<D: Dimension> {
```

Comment:
It makes sense to use `Dimension`, at least you simplify your life. But, if ever we'll find a way to convert `usize` in `Dimension`, it would be nice to also label this one with a `usize`.

Reply:
Yes... either dimension to usize or usize to dimension will simplify things in a few places but I don't know how to do so.

Reply:
Nor myself, I hope I will have some more time to investigate...

Reply:
Have a look at commit c50c07a34488b1f379897d8397c618c221b64136. Is this what you had in mind? The `where` clauses are a bit unfortunate, but they are needed because `Grid<D>` isn't defined for all `D`, currently only for `1` and `2`.

Reply:
Yes! This is actually great because it might also allow me to do some other stuff I wanted to do and didn't know how.